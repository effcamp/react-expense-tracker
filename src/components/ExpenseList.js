import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import { removeExpense } from '../actions/expenses';
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No expenses</p>
    ) : (
      props.expenses.map((expense) => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
    {}
  </div>
);

// check this out later
const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
