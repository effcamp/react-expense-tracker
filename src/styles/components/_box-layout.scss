.box-layout {
  background: url('/images/bg.jpg');
  background-size: cover;
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  &__box {
    background: fade-out($color: white, $amount: 0.15);
    border-radius: 0.3rem;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
  }
  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}
