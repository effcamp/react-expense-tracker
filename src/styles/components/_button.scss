.btn {
  color: white;
  font-size: 1.8rem;
  background: $blue;
  padding: $s-size;
  border: none;
  font-weight: 300;
  display: inline-block;
  text-decoration: none;
  line-height: 1;

  &--link {
    background: none;
  }
}
