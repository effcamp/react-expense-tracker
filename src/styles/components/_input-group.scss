.input-group {
  display: flex;
  flex-direction: column;
  margin: $m-size;

  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
    flex-direction: row;
  }

  &__item {
    margin-bottom: $s-size;

    @media (min-width: $desktop-breakpoint) {
      margin: 0 $s-size 0 0;
    }
  }
}
