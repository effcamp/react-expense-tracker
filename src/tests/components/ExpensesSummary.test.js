import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should render ExpensesSummary with 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={1} expensesTotal={235} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary with many expenses', () => {
  const wrapper = shallow(
    <ExpensesSummary expensesCount={22} expensesTotal={1239087} />
  );
  expect(wrapper).toMatchSnapshot();
});
