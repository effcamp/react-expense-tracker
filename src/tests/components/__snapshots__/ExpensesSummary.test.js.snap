// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $235.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary with many expenses 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $1,239,087.00
  </h1>
</div>
`;
